Upgrade go deps

mfridman opened the pull request.

mfridman commented:
Need to update golangci-lint

k2589 commented:
Do you need any help?

mfridman commented:
I merged the dependabot upgrades manually and will deal with golangci-lient when the dust settles.

ippoippo commented:
Hello @mfridman 
Sorry to pester. Do you have an ETA on when you expect to work on this (or is there an external dependency issue you are waiting on).

Reason I'm asking is so that we can plan around your next tagged release.

Thanks 🙇

mfridman commented:
I'll do it tonight, so let's say within the next 24 hours.

ippoippo commented:
> I'll do it tonight, so let's say within the next 24 hours.

Ohh. Thanks. Really appreciate the prompt response 🙇

mfridman commented:
Cut a new release, updated golangci-lint, cleaned up the code and resolved the security issues.

https://github.com/pressly/goose/releases/tag/v3.24.2
